webapp/src/routes/app/+page.svelte

```diff
@@ -1,53 +1,221 @@
+        document
+            .getElementById(`listing-${listingId}`)
+            ?.scrollIntoView({ behavior: 'smooth', block: 'center' });
+    };
+
+    const scrollToTop = () => {
+        if (listingsContainer) {
+            listingsContainer.scrollTo({ top: 0, behavior: 'smooth' });
+        }
+    };
+
+    const handleScroll = () => {
+        showBackToTop = listingsContainer.scrollTop > 0;
+    };
 </script>
 
 <Navbar />
 
-<body>
-    <div class="container">
-        <div class="listings">
-            <h2>Listings go here!</h2>
-        </div>
-        <div class="map-view">
-            <MapLibre
-                center={[-97.1, 49.9]}
-                zoom={10}
-                class="map"
-                standardControls
-                style="https://basemaps.cartocdn.com/gl/positron-gl-style/style.json"
-            />
-        </div>
+<div class="container">
+    <div class="listings" bind:this={listingsContainer}>
+        <Search
+            class="search-input"
+            type="text"
+            bind:value={searchQuery}
+            placeholder="Explore spots around this area..."
+            name="search"
+            on:input={() => {
+                dropdownOpen = true;
+            }}
+        />
+
+        {#if results.length > 0 && dropdownOpen}
+            <ul class="dropdown">
+                {#each results as result}
+                    <!-- svelte-ignore a11y-click-events-have-key-events a11y-no-noninteractive-element-interactions -->
+                    <li on:click={() => handleSelect(result)}>
+                        {result.properties.formatted}
+                    </li>
+                {/each}
+            </ul>
+        {/if}
+
+        {#each spots_data as listing}
+            <div
+                class="booking-info-container {listing.id === selectedListingId ? 'highlight' : ''}"
+                id={`listing-${listing.id}`}
+            >
+                <SpotsListComponent {listing} />
+            </div>
+        {/each}
+
+        {#if showBackToTop}
+            <button class="back-to-top" on:click={scrollToTop}> ↑ Back to Top </button>
+        {/if}
+    </div>
+
+    <div class="map-view">
+        <MapLibre
+            center={mapCenter}
+            {zoom}
+            style="https://basemaps.cartocdn.com/gl/positron-gl-style/style.json"
+        >
+            <GeolocateControl position="bottom-left" fitBoundsOptions={{ maxZoom: 12 }} />
+            {#each spots_data as { id, location }}
+                <DefaultMarker
+                    lngLat={[location.longitude, location.latitude]}
+                    on:click={() => handleMarkerClick(id)}
+                >
+                    <Popup offset={[0, -10]}>
+                        <div class="text-lg font-bold">
+                            {location.street_address}, {location.city}
+                        </div>
+                        <button on:click={() => handleMarkerClick(id)}>Go to listing</button>
```

Comment:
why do we need 2 handleMarkerClicks here?

Reply:
I was planning to do the highlighting when the marker is clicked but later changed it, I will remove the first one.